packages/chopsticks/src/blockchain/block-builder.ts

```diff
@@ -153,17 +154,45 @@ export const buildBlock = async (
   const pendingExtrinsics: HexString[] = []
   const includedExtrinsic: HexString[] = []
 
+  // apply ump via storage override hack
+  if (Object.keys(ump).length > 0) {
+    const meta = await head.meta
+    const layer = head.pushStorageLayer()
+    for (const [paraId, upwardMessages] of Object.entries(ump)) {
+      const queueSize = meta.registry.createType('(u32, u32)', [
+        upwardMessages.length,
+        upwardMessages.map((x) => x.length).reduce((s, i) => s + i, 0),
+      ])
+
+      const messages = meta.registry.createType('Vec<Bytes>', upwardMessages)
+
+      // TODO: make sure we append instead of replace
+      layer.setAll([
+        [compactHex(meta.query.ump.relayDispatchQueues(paraId)), messages.toHex()],
+        [compactHex(meta.query.ump.relayDispatchQueueSize(paraId)), queueSize.toHex()],
+      ])
+    }
+
+    logger.trace(
+      {
+        number: newBlock.number,
+        tempHash: newBlock.hash,
+        ump,
+      },
+      'Upward messages'
+    )
+
+    const needsDispatch = meta.registry.createType('Vec<u32>', Object.keys(ump))
+    layer.set(compactHex(meta.query.ump.needsDispatch()), needsDispatch.toHex())
+  }
+
   // apply extrinsics
   for (const extrinsic of extrinsics) {
     try {
       const { result, storageDiff } = await newBlock.call('BlockBuilder_apply_extrinsic', [extrinsic])
       const outcome = registry.createType<ApplyExtrinsicResult>('ApplyExtrinsicResult', result)
       if (outcome.isErr) {
-        if (outcome.asErr.isInvalid && outcome.asErr.asInvalid.isFuture) {
-          pendingExtrinsics.push(extrinsic)
```

Comment:
why removing this?

Reply:
it only makes tests harder to write. I don't see why we need this feature